pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -343,11 +343,6 @@ func (asw *actualStateOfWorld) SetVolumeMountedByNode(
 	if mounted {
 		// Increment set count
 		nodeObj.mountedByNodeSetCount = nodeObj.mountedByNodeSetCount + 1
-	} else {
-		// Do not allow value to be reset unless it has been set at least once
-		if nodeObj.mountedByNodeSetCount == 0 && !forceUnmount {
-			return nil
-		}
```

Comment:
what is the repurcussion of this ?

Reply:
The repurcussion of removing the check is, node has ultimate say in - "whether a volume is mounted or not". Earlier we used to discard the update from node until node said disk is mounted at least once. Now we always rely on node's value and hence disks that aren't mounted on the node can be detached sooner.